src/ORM/Association/BelongsToMany.php

```diff
@@ -510,7 +510,7 @@ protected function _appendNotMatching($query, $options)
         $subquery = $this->_appendJunctionJoin($subquery, $conditions);
 
         $query
-            ->andWhere(function ($exp) use ($subquery, $conds) {
+            ->where(function ($exp) use ($subquery, $conds) {
```

Comment:
This was actually on purpose, because of the derp I made with `$query->orWhere()` leaving the query in `OR` mode. To force it out of that mode, I used `andWhere`